backend/tests/integration/tests/test_deployments.py

```diff
@@ -2378,16 +2413,157 @@ def do_test_show_artifact_size(self, clean_mongo, user_token, devs):
+            artifacts, test_artifacts[2:4]
+        )
+        
+        # search for names with prefix and exact name
+        r = api_dep_v2.with_auth(user_token).call(
+            "GET", deployments_v2.URL_ARTIFACTS +
+            "?name=deployments-phase-testing&name=test-artifact-*"
+        )
+        assert r.status_code == 400
+
+        # search for multiple prefix
+        r = api_dep_v2.with_auth(user_token).call(
+            "GET", deployments_v2.URL_ARTIFACTS +
+            "?name=test*&name=test-artifact-*"
+        )
+        assert r.status_code == 400
+
+        # search for exact description 
+        r = api_dep_v2.with_auth(user_token).call(
+            "GET", deployments_v2.URL_ARTIFACTS +
+            "?description=abc"
+        )
+        assert r.status_code == 200
+        artifacts = r.json()
+        assert compare_artifact_names(
+            artifacts, test_artifacts[0:2]
+        )
+
+        # search for description with prefix
+        r = api_dep_v2.with_auth(user_token).call(
+            "GET", deployments_v2.URL_ARTIFACTS +
+            "?description=abc-*"
+        )
+        assert r.status_code == 200
+        artifacts = r.json()
+        assert compare_artifact_names(
+            artifacts, test_artifacts[2:4]
+        )
+
+        # search for exact device type
+        r = api_dep_v2.with_auth(user_token).call(
+            "GET", deployments_v2.URL_ARTIFACTS +
+            "?device_type=arm1"
+        )
+        assert r.status_code == 200
+        artifacts = r.json()
+        assert compare_artifact_names(
+            artifacts, test_artifacts[1:2]
+        )
+        
+        # search for device type with prefix
+        r = api_dep_v2.with_auth(user_token).call(
+            "GET", deployments_v2.URL_ARTIFACTS +
+            "?device_type=arm1-*"
+        )
+        assert r.status_code == 200
+        artifacts = r.json()
+        assert compare_artifact_names(
+            artifacts, test_artifacts[2:4]
+        )
+
+        # test pagination
+        r = api_dep_v2.with_auth(user_token).call(
+            "GET", deployments_v2.URL_ARTIFACTS +
+            "?page=2&per_page=3"
+        )
+        assert r.status_code == 200
+        artifacts = r.json()
+        assert compare_artifact_names(
+            artifacts, test_artifacts[3:4]
+        )
+
+
 
 @pytest.mark.storage_test
-class TestDeploymentShowArtifactSizeOpenSource(_TestDeploymentsShowArtifactSizeBase):
+class TestDeploymentArtifactOpenSource(_TestDeploymentsArtifactBase):
     def test_show_artifact_size(self, clean_mongo):
         _user, user_token, devs = setup_devices_and_management_st(10)
         self.do_test_show_artifact_size(clean_mongo, user_token, devs)
+    def test_list_artifacts_v2(self, clean_mongo):
+        _user, user_token, devs = setup_devices_and_management_st()
+        artifacts = create_test_artifacts(utoken=user_token)
+        artifacts.insert(0, "deployments-phase-testing")
```

Comment:
Not very obvious where this comes in, but figured out it comes as a side-effect of `setup_devices_and_management`...